### PR TITLE
Add config set support

### PR DESCRIPTION
**Pull request by roborourke**
Related to https://github.com/humanmade/product-dev/issues/822

### CODE REVIEW

File: inc/class-ses.php [hunk 1]
```diff
@@ -20,20 +21,22 @@ public static function get_instance() {
 
 		if ( ! self::$instance ) {
 
-			$key    = defined( 'AWS_SES_WP_MAIL_KEY' ) ? AWS_SES_WP_MAIL_KEY : null;
-			$secret = defined( 'AWS_SES_WP_MAIL_SECRET' ) ? AWS_SES_WP_MAIL_SECRET : null;
-			$region = defined( 'AWS_SES_WP_MAIL_REGION' ) ? AWS_SES_WP_MAIL_REGION : null;
+			$key    	= defined( 'AWS_SES_WP_MAIL_KEY' ) ? AWS_SES_WP_MAIL_KEY : null;
+			$secret 	= defined( 'AWS_SES_WP_MAIL_SECRET' ) ? AWS_SES_WP_MAIL_SECRET : null;
+			$region 	= defined( 'AWS_SES_WP_MAIL_REGION' ) ? AWS_SES_WP_MAIL_REGION : null;
+			$config_set = defined( 'AWS_SES_WP_MAIL_CONFIG_SET' ) ? AWS_SES_WP_MAIL_CONFIG_SET : null;
```
Comment (roborourke) [inline inc/class-ses.php:27]: We don't bother with aligning assignment operators any more as it creates unnecessary code churn:
```suggestion
			$key = defined( 'AWS_SES_WP_MAIL_KEY' ) ? AWS_SES_WP_MAIL_KEY : null;
			$secret = defined( 'AWS_SES_WP_MAIL_SECRET' ) ? AWS_SES_WP_MAIL_SECRET : null;
			$region = defined( 'AWS_SES_WP_MAIL_REGION' ) ? AWS_SES_WP_MAIL_REGION : null;
			$config_set = defined( 'AWS_SES_WP_MAIL_CONFIG_SET' ) ? AWS_SES_WP_MAIL_CONFIG_SET : null;
```

Also if you are aligning things pay close attention to tabs vs spaces, alignment is best done using spaces.

File: inc/class-ses.php [hunk 2]
```diff
@@ -20,20 +21,22 @@ public static function get_instance() {
...
 
 			self::$instance = new static( $key, $secret, $region );
```
Comment (roborourke) [inline inc/class-ses.php:29]: Should `$config_set` be passed as the 4th parameter here?

File: inc/class-ses.php [hunk 3]
```diff
@@ -20,20 +21,22 @@ public static function get_instance() {
...
 		}
 
 		return self::$instance;
 	}
 
-	public function __construct( $key, $secret, $region = null ) {
-		$this->key = $key;
-		$this->secret = $secret;
-		$this->region = $region;
+	public function __construct( $key, $secret, $region = null, $config_set = "" ) {
```
Comment (roborourke) [inline inc/class-ses.php:35]: The fallback value for `$region` is `null`, and the fallback for the config set constant is `null` so lets keep this consistent.
```suggestion
	public function __construct( $key, $secret, $region = null, $config_set = null ) {
```

File: inc/class-ses.php [hunk 4]
```diff
@@ -20,20 +21,22 @@ public static function get_instance() {
...
+		$this->key 		  = $key;
+		$this->secret 	  = $secret;
+		$this->region 	  = $region;
+		$this->config_set = $config_set;
```
Comment (roborourke) [inline inc/class-ses.php:39]:
```suggestion
		$this->key = $key;
		$this->secret = $secret;
		$this->region = $region;
		$this->config_set = $config_set;
```

File: inc/class-ses.php [hunk 5]
```diff
@@ -158,6 +161,7 @@ public function send_wp_mail( $to, $subject, $message, $headers = [], $attachmen
 					],
 					'Body'   => [],
 				],
+				'ConfigurationSetName' => $this->config_set,
 			];
```
Comment (roborourke) [inline inc/class-ses.php:165]: What values are acceptable here? If the API rejects a `null` value or empty string we can set this in an additive way e.g:

```suggestion
			];

			if ( ! empty( $this->config_set ) ) {
				$args['ConfigurationSetName'] = $this->config_set;
			}
```

**Comment by roborourke** [comment]
@ferschubert-hm I've reviewed - you can batch apply the changes via the files tab on this PR then see if solves the errors you noted on the related issue.

**Comment by roborourke** [comment]
@ferschubert-hm if this gets things working you can approve the PR and merge it in. I can handle making a release. Will do it as a minor version update.

**Review by ferschubert-hm — APPROVED** [review]

**Comment by ferschubert-hm** [comment]
Thanks @roborourke changes fixed it. Tested in platform-test. I have added just some updates to the README.md about this new feature and will merge now the changes.

**Comment by roborourke** [comment]
Released in 1.3.0

**Pull request merged**